test/jdk/javax/swing/JRadioButton/ButtonGroupFocus/ButtonGroupFocusTest.java

```diff
@@ -69,52 +100,104 @@ public static void main(String[] args) throws Exception {
             group.add(button4);
             group.add(button5);
 
+            button2.addFocusListener(new LatchFocusListener(button2FocusLatch));
+            button3.addFocusListener(new LatchFocusListener(button3FocusLatch));
+            button4.addFocusListener(new LatchFocusListener(button4FocusLatch));
+
+            button2.addFocusListener(new LatchFocusListener(button2FocusLatch2));
+
             button2.setSelected(true);
 
+            // Debugging aid: log focus owner changes...
+            KeyboardFocusManager focusManager = getCurrentKeyboardFocusManager();
+            focusManager.addPropertyChangeListener("focusOwner",
+                    new PropertyChangeListener() {
+                        @Override
+                        public void propertyChange(PropertyChangeEvent evt) {
+                            System.out.println(evt.getPropertyName()
+                                               + "\n\t" + evt.getOldValue()
+                                               + "\n\t" + evt.getNewValue());
+                        }
+                    });
```

Comment:
```suggestion
                    evt -> System.out.println(evt.getPropertyName()
                                               + "\n\t" + evt.getOldValue()
                                               + "\n\t" + evt.getNewValue()));
```

Nit. Possible lambda replacement here.

Reply:
My IDE didn't suggest replacing it a lambda. Does it improve readability of the code? Perhaps, it does… by reducing the number of lines.

Reply:
Not by much if it does. I got into the habit of looking for where it's possible since reducing the line count usually makes it more readable.